pkg/kubectl/cmd/logs.go

```diff
@@ -191,22 +194,50 @@ func (o *LogsOptions) Complete(f cmdutil.Factory, out io.Writer, cmd *cobra.Comm
 		builder := resource.NewBuilder(mapper, typer, o.ClientMapper, decoder).
 			NamespaceParam(o.Namespace).DefaultNamespace().
 			SingleResourceType()
-		if o.ResourceArg != "" {
-			builder.ResourceNames("pods", o.ResourceArg)
-		}
-		if selector != "" {
-			builder.ResourceTypes("pods").SelectorParam(selector)
-		}
-		infos, err := builder.Do().Infos()
-		if err != nil {
-			return err
-		}
-		if selector == "" && len(infos) != 1 {
-			return errors.New("expected a resource")
+		if o.ResourceArg != "" && selector == "" {
+			infos, err := builder.ResourceTypeOrNameArgs(true, "pods").
+				Flatten().Do().Infos()
+			if err != nil {
+				return err
+			}
+			//match pods using prefix
+			prefixTimes := 0
+			for i, info := range infos {
+				if strings.HasPrefix(info.Name, o.ResourceArg) {
+					o.PrefixMatchList = append(o.PrefixMatchList, info.Name)
+					o.Object = infos[i].Object
+					prefixTimes++
+				}
+			}
+			if prefixTimes == 0 {
```

Comment:
If the prefix given by the user as a podName did not match any pods in the loop above, wouldn't the builder below fail to match any pods as well using the same `o.ResourceArg` value?